platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/DefaultConfigurationCache.kt

```diff
@@ -204,6 +203,7 @@ class DefaultConfigurationCache internal constructor(
             // Can reuse the cache entry for the rest of this build invocation
             cacheAction = ConfigurationCacheAction.LOAD
         }
+        scopeRegistryListener.dispose()
```

Comment:
This is already called by `finalizeCacheEntry` which happens around the same time so I'd just remove this call from here.

Reply:
@bamboo Could you confirm the further : Consider we have a CC hit. We are having some load of models/workgraph, does `finalizeCacheEntry` will be even invoked?

Reply:
When loading from the cache, the listener is already disposed earlier in https://github.com/gradle/gradle/blob/5ff049685f7bbd5c5ec3b2f493bdafa72ad4c7ef/platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/DefaultConfigurationCache.kt#L387

And that makes me think, can Gradle get into a situation where one worker is computing a model (and thus might need the listener intact) while another worker is disposing of the listener after having loaded a model from the cache? 🤔

Reply:
It seems like your initial comment was left on the line in `finalizeCacheEntry`, and this is not something that you wanted?

Reply:
Yes, I got confused with the diff interface. Looks good 👍

Reply:
> can Gradle get into a situation where one worker is computing a model (and thus might need the listener intact) while another worker is disposing of the listener after having loaded a model from the cache?

FTR: No, since `finalizeCacheEntry` is called once after everything has been written to the cache.